impl/src/bin/cargo-raze.rs

```diff
@@ -65,44 +69,60 @@ Options:
 "#;
 
 fn main() {
-  let mut config = Config::default().unwrap();
-
   let options = Docopt::new(USAGE)
     .and_then(|d| d.deserialize())
     .unwrap_or_else(|e| e.exit());
 
-  let result = real_main(&options, &mut config);
+  let metadata_fetcher = create_fetcher(&options);
+  let result = real_main(&options, metadata_fetcher);
+
+  if let Err(_e) = result {
+    std::process::exit(-1);
+  }
+}
+
+#[cfg(cargo_internals)]
+fn create_fetcher(options: &Options) -> Box<dyn MetadataFetcher> {
+  create_cargo_internals_fetcher(options)
+}
+
+#[cfg(not(cargo_internals))]
+fn create_fetcher(options: &Options) -> Box<dyn MetadataFetcher> {
+  create_subcommand_fetcher(options)
+}
 
-  if let Err(e) = result {
-    cargo::exit_with_error(e, &mut *config.shell());
+#[cfg(cargo_internals)]
+fn create_cargo_internals_fetcher(options: &Options) -> Box<dyn MetadataFetcher> {
```

Comment:
Can this function body be inlined into the create_fetcher for cargo_internals?

As is, the name is misleading (we might get a subcommand fetcher).